account_move_product/models/account_move.py

```diff
@@ -0,0 +1,12 @@
+# Copyright 2022 Quartile Limited
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    product_id = fields.Many2one(
+        "product.product", related="invoice_line_ids.product_id", string="Product"
+    )
```

Comment:
In 16.0, there is search on `line_ids` (not in 15.0) and I think we may just take that approach.  I think that should suffice.  Let's remove this field and just adjust the search view.  Please change the module name to `account_move_search_line` and propose the module in OCA as appropriate.

![image](https://github.com/qrtl/sst-custom/assets/7766939/d12abf93-c575-4f1d-a8fa-b5090153469b)

Reply:
Now that this PR is trying to move to the **15.0** version, line_ids cannot be used?

Reply:
invoice_line_ids is also exit in v15.0 and we can also use for search. https://github.com/odoo/odoo/blob/4ee1def9c11021e1f692ab9cb769005aa2fbdbd0/addons/account/models/account_move.py#L316.